crates/nu-cli/src/commands/classified/block.rs

```diff
@@ -40,30 +40,25 @@ pub async fn run_block(
                             inp,
                         )
                         .await?;
-                    loop {
-                        match output_stream.try_next().await {
-                            Ok(Some(ReturnSuccess::Value(Value {
-                                value: UntaggedValue::Error(e),
-                                ..
-                            }))) => return Err(e),
-                            Ok(Some(_item)) => {
-                                if let Some(err) = ctx.get_errors().get(0) {
-                                    ctx.clear_errors();
-                                    return Err(err.clone());
-                                }
-                                if ctx.ctrl_c.load(Ordering::SeqCst) {
-                                    break;
```

Comment:
Possibly this was intending to break from the outer loop, and this `break` statement should remain? In the existing version of the code, the statement has no function IIUC, given that the inner loop doesn't loop

Reply:
Right, it's to break from the loop. Since streams can be infinite, you may be pulling on an infinite stream's values, and this is one of the places that hitting ctrl-c should break you out of pulling on that stream.

Reply:
Yes that's clear. 

Is my suggestion that the existing code wasn't doing that — because the existing `break` applied to an [inner loop that never looped anyway](https://github.com/nushell/nushell/pull/2830/files#diff-46a92fe54607f3265ff680b05118c4a1c863ac6308d4ceddb232ff225009de04L43) — correct?

Reply:
Oh because it's the output of autoview? Yeah, you're probably right that this wouldn't loop.